pkg/server/server.go

```diff
@@ -102,3 +114,18 @@ func New(configuration options.Configuration) (controllerruntime.Manager, error)
 
 	return mgr, nil
 }
+
+func k8sCommunicationCheck(client *discovery.DiscoveryClient) (err error) {
+	// retry 6 times with a delay of 5 seconds each retry
+	for range [6]int{} {
```

Comment:
query: How did we arrive at these numbers i.e. 6*5 = 30seconds timeout

Reply:
I just picked a number I thought was reasonable. anecdotally I have seen the sidecar take up to ~15/20 seconds at times. I'm open to suggestions 👍  

@AmitKumarDas  thanks for the feedback!

Reply:
IMO lets put these observations/numbers in the code comments as well since you have already mentioned service mesh.